modules/tinymce/src/core/main/ts/init/InitContentBody.ts

```diff
@@ -269,16 +270,36 @@ const initEditor = function (editor: Editor) {
 const getStyleSheetLoader = (editor: Editor): StyleSheetLoader =>
   editor.inline ? editor.ui.styleSheetLoader : editor.dom.styleSheetLoader;
 
+const makeStylesheetLoadingPromises = (editor: Editor, css: string[], framedFonts: string[]): Promise<unknown>[] => {
+  const promises = [
+    new Promise((resolve, reject) => getStyleSheetLoader(editor).loadAll(css, resolve, reject)),
+  ];
+
+  if (editor.inline) {
+    return promises;
+  } else {
+    return promises.concat([
+      new Promise((resolve, reject) => editor.ui.styleSheetLoader.loadAll(framedFonts, resolve, reject)),
+    ]);
+  }
+};
+
 const loadContentCss = (editor: Editor, css: string[]) => {
   const styleSheetLoader = getStyleSheetLoader(editor);
 
   const loaded = () => {
-    editor.on('remove', () => styleSheetLoader.unloadAll(css));
+    editor.on('remove', () => {
+      styleSheetLoader.unloadAll(css);
+
+      if (!editor.inline) {
+        editor.ui.styleSheetLoader.unloadAll(Settings.getFontCss(editor));
```

Comment:
Doing `Settings.getFontCss(editor)` again is dangerous, as the setting can be mutated. It's also duplicate code, so we should store the original value here.